shark/iree_utils/gpu_utils.py

```diff
@@ -68,15 +68,23 @@ def get_rocm_device_arch(device_num=0, extra_args=[]):
     arch_in_device_dump = None
 
     # get rocm arch from iree dump devices
-    def get_devices_info_from_dump(dump):
+    def get_devices_info_from_dump(dump, driver):
         from os import linesep
-
-        dump_clean = list(
-            filter(
-                lambda s: "--device=rocm" in s or "gpu-arch-name:" in s,
-                dump.split(linesep),
+        
+        if driver == "hip":
+            dump_clean = list(
+                filter(
+                    lambda s: "AMD" in s,
+                    dump.split(linesep),
+                )
+            )
+        else:
+            dump_clean = list(
+                filter(
+                    lambda s: f"--device={driver}" in s or "gpu-arch-name:" in s,
+                    dump.split(linesep),
+                )
             )
-        )
         arch_pairs = [
             (
                 dump_clean[i].split("=")[1].strip(),
```

Comment:
Double check this works for hip devices. I had to rewrite it a bit to make it work locally, I can post that if need be.

Reply:
Gave me the right output for 7900xtx through parse_device, please follow up if you see any issues